core/src/nodes/collection.rs

```diff
@@ -439,6 +439,36 @@ impl<TInEvent, TOutEvent, THandler> CollectionStream<TInEvent, TOutEvent, THandl
     }
 }
 
+/// Reach attempt interrupt errors. 
+#[derive(Debug, PartialEq)]
+pub enum InterruptError {
+    /// An invalid reach attempt has been used to try to interrupt. The task
+    /// entry is vacant; it needs to be added first via add_reach_attempt
+    /// (with the TaskState set to Pending) before we try to connect.
+    ReachAttemptNotFound,
+    /// The task has already connected to the node; interrupting a reach attempt
+    /// is thus redundant as it has already completed. Thus, the reach attempt
+    /// that has tried to be used is no longer valid, since already reached.
+    AlreadyReached,
+}
+
+impl fmt::Display for InterruptError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            InterruptError::ReachAttemptNotFound => 
+                write!(f, "The reach attempt could not be found."),
+            InterruptError::AlreadyReached =>
+                write!(f, "The reach attempt has already completed or reached the node."),
+        }
+    }
+}
+
+impl error::Error for InterruptError {
```

Comment:
I don't think you need more than `impl Error for InterruptError {}` here (will use the default impls for the trait's methods and you have already covered the `Debug`/`Display` supertraits).

Reply:
OK, all done.